corporate/lib/stripe.py

```diff
@@ -551,6 +633,35 @@ def get_data_for_stripe_customer(self) -> StripeCustomerData:
         )
         return realm_stripe_customer_data
 
+    @override
+    def update_data_for_checkout_session_and_payment_intent(
+        self, metadata: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        updated_metadata = dict(
+            user_email=self.user.delivery_email,
+            realm_id=self.realm.id,
+            realm_str=self.realm.string_id,
+            user_id=self.user.id,
+            **metadata,
+        )
+        return updated_metadata
+
+    @override
+    def get_data_for_stripe_payment_intent(
+        self, price_per_license: int, licenses: int
+    ) -> StripePaymentIntentData:
+        # Support requests do not set any stripe billing information.
+        assert self.support_session is False
+        amount = price_per_license * licenses
+        description = f"Upgrade to Zulip Cloud Standard, ${price_per_license/100} x {licenses}"
+        plan_name = "Zulip Cloud Standard"
+        return StripePaymentIntentData(
+            amount=amount,
+            description=description,
+            plan_name=plan_name,
+            email=self.user.delivery_email,
+        )
```

Comment:
This is a useful intermediate checkpoint but I think this function will want to take another form:
* A `self.contact_email()` property should probably be used to get the email to use for contacting that user?
* We probably want to add a `tier` parameter rather than having "Zulip Cloud Standard" be hardcoded.

But those can readily be done in a next pass; just needs coordination with @amanagr.